src/reconcile/rec_visibility.c

```diff
@@ -471,8 +471,13 @@ __rec_validate_upd_chain(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE *s
             prev_upd->start_ts == prev_upd->durable_ts || !WT_TIME_WINDOW_HAS_STOP(&vpack->tw) ||
             prev_upd->durable_ts >= vpack->tw.durable_stop_ts,
           "Durable timestamps cannot be out of order for prepared updates");
-        if (prev_upd->start_ts < vpack->tw.start_ts ||
-          (WT_TIME_WINDOW_HAS_STOP(&vpack->tw) && prev_upd->start_ts < vpack->tw.stop_ts)) {
+        /*
+         * If prev_upd is globally visible it can be evicted. Thus this check applies only if
```

Comment:
evicted is not the right word. I think it should be the following updates can be removed by obsolete check.

Reply:
Why is evicted not the right word?  That is the word used throughout this function. I will talk with you about this.
```
 381      * If eviction reconciliation starts before checkpoint, it is fine to evict updates without
 382      * timestamps.

 389         WT_STAT_CONN_DATA_INCR(session, cache_eviction_blocked_remove_hs_race_with_checkpoint);

 401         WT_STAT_CONN_DATA_INCR(session, cache_eviction_blocked_no_ts_checkpoint_race_2);

 433             WT_STAT_CONN_DATA_INCR(session, cache_eviction_blocked_no_ts_checkpoint_race_4);

 475          * If prev_upd is globally visible it can be evicted. Thus this check applies only if
 476          * prev_upd is not globally visible.

 482             WT_STAT_CONN_DATA_INCR(session, cache_eviction_blocked_no_ts_checkpoint_race_1);
```

Reply:
obsolete check may happen at any time not necessarily in eviction.

Reply:
Rollback to stable may recover updates from the history store that is out of order to the on-disk value. Normally these updates have the WT_UPDATE_RESTORED_FROM_HS flag on them. However, in rare cases, if the newer update becomes globally visible. The restored update may be removed by the obsolete check. This may lead to an out of order edge case but it is benign. Check the global visibility of the update and ignore this case.

Reply:
Done